### ISSUE TITLE
[Bug] Using inside a NodeJS project - error TS2300: Duplicate identifier 'beforeEach'

### ISSUE THREAD

**Issue by Dzhuneyt** [havsar/node-ts-cache#5]
Including this inside my TypeScript-based NodeJS project, I started receiving the errors below. After some digging around, I noticed that @types/mocha is declared as a dependency of this project.

Is it actually needed for some important purposes? I'm sure it's not a blocking point for many of the users of this library since they'll not use it in the same context that I do, but still, somebody somewhere may experience the same. Any chance we could drop that dependency?

```

> backend@1.0.0 tsc /home/ubuntu/www/Private/RSS-News-Feed-Portal/backend
> node ./node_modules/typescript/bin/tsc

node_modules/@types/mocha/index.d.ts:2680:13 - error TS2300: Duplicate identifier 'beforeEach'.

2680 declare var beforeEach: Mocha.HookFunction;
                 ~~~~~~~~~~

  ../../../node_modules/@types/jasmine/index.d.ts:59:18
    59 declare function beforeEach(action: ImplementationCallback, timeout?: number): void;
                        ~~~~~~~~~~
    'beforeEach' was also declared here.
  ../../../node_modules/@types/jasminewd2/index.d.ts:13:18
    13 declare function beforeEach(action: (done: DoneFn) => Promise<void>, timeout?: number): void;
                        ~~~~~~~~~~
    and here.

node_modules/@types/mocha/index.d.ts:2698:13 - error TS2300: Duplicate identifier 'afterEach'.

2698 declare var afterEach: Mocha.HookFunction;
                 ~~~~~~~~~

  ../../../node_modules/@types/jasmine/index.d.ts:66:18
    66 declare function afterEach(action: ImplementationCallback, timeout?: number): void;
                        ~~~~~~~~~
    'afterEach' was also declared here.
  ../../../node_modules/@types/jasminewd2/index.d.ts:14:18
    14 declare function afterEach(action: (done: DoneFn) => Promise<void>, timeout?: number): void;
                        ~~~~~~~~~
    and here.

node_modules/@types/mocha/index.d.ts:2714:13 - error TS2300: Duplicate identifier 'describe'.

2714 declare var describe: Mocha.SuiteFunction;
                 ~~~~~~~~

  ../../../node_modules/@types/jasmine/index.d.ts:23:18
    23 declare function describe(description: string, specDefinitions: () => void): void;
                        ~~~~~~~~
    'describe' was also declared here.

node_modules/@types/mocha/index.d.ts:2735:13 - error TS2300: Duplicate identifier 'xdescribe'.

2735 declare var xdescribe: Mocha.PendingSuiteFunction;
                 ~~~~~~~~~

  ../../../node_modules/@types/jasmine/index.d.ts:25:18
    25 declare function xdescribe(description: string, specDefinitions: () => void): void;
                        ~~~~~~~~~
    'xdescribe' was also declared here.

node_modules/@types/mocha/index.d.ts:2749:13 - error TS2300: Duplicate identifier 'it'.

2749 declare var it: Mocha.TestFunction;
                 ~~

  ../../../node_modules/@types/jasmine/index.d.ts:34:18
    34 declare function it(expectation: string, assertion?: ImplementationCallback, timeout?: number): void;
                        ~~
    'it' was also declared here.
  ../../../node_modules/@types/jasminewd2/index.d.ts:10:18
    10 declare function it(expectation: string, assertion?: (done: DoneFn) => Promise<void>, timeout?: number): void;
                        ~~
    and here.

node_modules/@types/mocha/index.d.ts:2770:13 - error TS2300: Duplicate identifier 'xit'.

2770 declare var xit: Mocha.PendingTestFunction;
                 ~~~

  ../../../node_modules/@types/jasmine/index.d.ts:44:18
    44 declare function xit(expectation: string, assertion?: ImplementationCallback, timeout?: number): void;
                        ~~~
    'xit' was also declared here.
  ../../../node_modules/@types/jasminewd2/index.d.ts:12:18
    12 declare function xit(expectation: string, assertion?: (done: DoneFn) => Promise<void>, timeout?: number): void;
                        ~~~
    and here.

../../../node_modules/@types/jasmine/index.d.ts:23:18 - error TS2300: Duplicate identifier 'describe'.

23 declare function describe(description: string, specDefinitions: () => void): void;
                    ~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2714:13
    2714 declare var describe: Mocha.SuiteFunction;
                     ~~~~~~~~
    'describe' was also declared here.

../../../node_modules/@types/jasmine/index.d.ts:25:18 - error TS2300: Duplicate identifier 'xdescribe'.

25 declare function xdescribe(description: string, specDefinitions: () => void): void;
                    ~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2735:13
    2735 declare var xdescribe: Mocha.PendingSuiteFunction;
                     ~~~~~~~~~
    'xdescribe' was also declared here.

../../../node_modules/@types/jasmine/index.d.ts:34:18 - error TS2300: Duplicate identifier 'it'.

34 declare function it(expectation: string, assertion?: ImplementationCallback, timeout?: number): void;
                    ~~

  node_modules/@types/mocha/index.d.ts:2749:13
    2749 declare var it: Mocha.TestFunction;
                     ~~
    'it' was also declared here.

../../../node_modules/@types/jasmine/index.d.ts:44:18 - error TS2300: Duplicate identifier 'xit'.

44 declare function xit(expectation: string, assertion?: ImplementationCallback, timeout?: number): void;
                    ~~~

  node_modules/@types/mocha/index.d.ts:2770:13
    2770 declare var xit: Mocha.PendingTestFunction;
                     ~~~
    'xit' was also declared here.

../../../node_modules/@types/jasmine/index.d.ts:59:18 - error TS2300: Duplicate identifier 'beforeEach'.

59 declare function beforeEach(action: ImplementationCallback, timeout?: number): void;
                    ~~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2680:13
    2680 declare var beforeEach: Mocha.HookFunction;
                     ~~~~~~~~~~
    'beforeEach' was also declared here.

../../../node_modules/@types/jasmine/index.d.ts:66:18 - error TS2300: Duplicate identifier 'afterEach'.

66 declare function afterEach(action: ImplementationCallback, timeout?: number): void;
                    ~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2698:13
    2698 declare var afterEach: Mocha.HookFunction;
                     ~~~~~~~~~
    'afterEach' was also declared here.

../../../node_modules/@types/jasminewd2/index.d.ts:10:18 - error TS2300: Duplicate identifier 'it'.

10 declare function it(expectation: string, assertion?: (done: DoneFn) => Promise<void>, timeout?: number): void;
                    ~~

  node_modules/@types/mocha/index.d.ts:2749:13
    2749 declare var it: Mocha.TestFunction;
                     ~~
    'it' was also declared here.

../../../node_modules/@types/jasminewd2/index.d.ts:12:18 - error TS2300: Duplicate identifier 'xit'.

12 declare function xit(expectation: string, assertion?: (done: DoneFn) => Promise<void>, timeout?: number): void;
                    ~~~

  node_modules/@types/mocha/index.d.ts:2770:13
    2770 declare var xit: Mocha.PendingTestFunction;
                     ~~~
    'xit' was also declared here.

../../../node_modules/@types/jasminewd2/index.d.ts:13:18 - error TS2300: Duplicate identifier 'beforeEach'.

13 declare function beforeEach(action: (done: DoneFn) => Promise<void>, timeout?: number): void;
                    ~~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2680:13
    2680 declare var beforeEach: Mocha.HookFunction;
                     ~~~~~~~~~~
    'beforeEach' was also declared here.

../../../node_modules/@types/jasminewd2/index.d.ts:14:18 - error TS2300: Duplicate identifier 'afterEach'.

14 declare function afterEach(action: (done: DoneFn) => Promise<void>, timeout?: number): void;
                    ~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2698:13
    2698 declare var afterEach: Mocha.HookFunction;
                     ~~~~~~~~~
    'afterEach' was also declared here.


Found 16 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! backend@1.0.0 tsc: `node ./node_modules/typescript/bin/tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the backend@1.0.0 tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2019-02-15T14_10_04_123Z-debug.log

```

**Comment by havsar** [comment]
Your'e right, @types/mocha should be in the devDependencies. This will be fixed in the next minor version.

**Comment by havsar** [comment]
@Dzhuneyt Closed by d50e97ee1e9cb49a9e08c7ce1c6706124ab57810
